### ISSUE TITLE
Automatic naming of backups for advanced users

### ISSUE THREAD

**Issue by shasheene** [rescuezilla/rescuezilla#292]
Right now Rescuezilla autopopulates the backup name, continuing the approach of the old _Redo Backup and Recovery_ tool which the project was forked from.

A user on the Rescuezilla Sourceforge support forum [writes](https://sourceforge.net/p/rescuezilla/discussion/discussion/thread/ec74f2c4d3/?limit=25#f2aa)

>  My question/request, is there any way to have Rescuezilla do some type of automatic naming of the file? When using Clonezilla I am able to use [THESE PARAMETERS.](https://clonezilla.org/advanced/reserved-word-ocs-sr.php) 
> 
> At the same time, would it be possible to add a new type of name? I am saving images the naming convention of MAKE-MODEL-SERIAL-DATE. 
> 
> SERIAL being the output you get on a machine when you run this command: 
> 
> `sudo dmidecode -t system | grep Serial`
> 
> Again, thank you for all the help and thanks for any information on these questions.

My response:

> It's a reasonable request to be able to autogenerate the filename.
> 
> Rescuezilla v2.3 will be released at some point (I have rarely have large chunks of free time right now), but the release candidate I have developed (but not yet released) has the model and serial number listed in the drive information. But having the ability to place this information into the backup name automatically makes a lot of sense.
> 
> **I think the smartest and cleanest approach would allow the user to specify variables in the backup name field**, then display a preview.
>
> So the user to specify backup names like (for example) "%{model}-%{serial}-%{date}-laptop", and it would show a preview "laptop-WesternDigital-12345678-2021-12-07-laptop".
> 
> Would this be sufficient? It would be tedious to write out something like that every time. But easier than manually typing it.
>
> I can implement such a feature, but it's not a high-priority at this stage.

**Comment by compicat** [comment]
_**Rescuezilla v2.3 will be released at some point (I have rarely have large chunks of free time right now), but the release candidate I have developed (but not yet released) has the model and serial number listed in the drive information. But having the ability to place this information into the backup name automatically makes a lot of sense.**_

Maybe a tester group may help you to reduce the weight on your shoulders in this one man show. you can release rc version to be tested by several people -and of course- in several setups to get feedbacks. Just a suggestion.

**Comment by shasheene** [comment]
Yeah it's a reasonable request. But for now I think I just need to find two contiguous days to work on the project. I am confident I'll be able to make a release this weekend.

I might open up to human testers at some point, but right now I think improving the automated testing suite tooling will be the best approach. It will allow testing many thousands of images within a few hours, something that the project needs as I expect Rescuezilla to be useful for at least 50 years (Clonezilla has been around for 20 years so far, and for the 20 years before that there was Norton Ghost and other similar tools).
